server/src/main/java/org/opensearch/search/startree/filter/provider/DimensionFilterMapper.java

```diff
@@ -408,3 +411,49 @@ private Object parseRawKeyword(String field, Object rawValue, KeywordFieldType k
     }
 
 }
+
+class StarDateFieldMapper implements DimensionFilterMapper {
+
+    @Override
+    public DimensionFilter getExactMatchFilter(MappedFieldType mappedFieldType, List<Object> rawValues) {
+        if (mappedFieldType instanceof DateFieldMapper.DateFieldType) {
+            DateFieldMapper.DateFieldType dateFieldType = (DateFieldMapper.DateFieldType) mappedFieldType;
+            List<Object> convertedValues = new ArrayList<>(rawValues.size());
+            for (Object rawValue : rawValues) {
+                convertedValues.add(dateFieldType.parse(rawValue.toString()));
+            }
+            return new ExactMatchDimFilter(mappedFieldType.name(), convertedValues);
+        }
+        return null;
+    }
+
+    // This is unused as we need range query builder to compute this filter for date fields
```

Comment:
The interface definition was suiting keyword and numeric field types too well, so it clearly missed out on date type. 

Can we change the interface to pass the RangeQueryBuilder itself ? 

Spilling out the logic of data type to StarTreeFilterMapper would mean all the parsing related to a field type is not contained within the DimensionFilterMapper